### PR TITLE
feat: add hasKeyboardFocus as attribute

### PR DESCRIPTION
**Pull request by KazuCocoa**

### CODE REVIEW

File: WebDriverAgentLib/Routing/FBXCElementSnapshotWrapper.m [hunk 1]
```diff
@@ -94,12 +94,19 @@ - (BOOL)isSelected
...
 - (BOOL)hasFocus
 {
   return self.snapshot.hasFocus;
 }
+
 #endif
 
+- (BOOL)hasKeyboardFocus
+{
+  return self.snapshot.hasKeyboardFocus;
```
Comment (mykola-mokhnach) [inline WebDriverAgentLib/Routing/FBXCElementSnapshotWrapper.m:107]: I assume internally hasFocus and hasKeyboardFocus are the same, aren't they?

Comment (KazuCocoa) [reply WebDriverAgentLib/Routing/FBXCElementSnapshotWrapper.m:107]: Hm, the result was different in iOS. I haven't checked on tvOS.
For example when i selected a search bar (then keyboard appeared), the hasFocus was false while the hasKeyboardFocus was true for the element. (I thought they were the same)

Comment (mykola-mokhnach) [reply WebDriverAgentLib/Routing/FBXCElementSnapshotWrapper.m:107]: ok, how about tvOS then? Does this attribute make any sense there?
Also, the hasKeyboardFocus property does not seem to be public. I am not sure we want to expose it as Apple may change/remove it unexpectedly and then things gonna go complicated on our side

Comment (KazuCocoa) [reply WebDriverAgentLib/Routing/FBXCElementSnapshotWrapper.m:107]: tvOS case was opposit. A focused element had hasFocused true but hasKeyboardFocus was false. (tested only simulator tho)

Comment (KazuCocoa) [reply WebDriverAgentLib/Routing/FBXCElementSnapshotWrapper.m:107]: hm, yea, `hasKeyboardFocus` is a private header only for now. okay, lets close this pr. hopefully the hasFocus will work in ios as well.

**Comment by KazuCocoa** [comment]
Close as not public api doc

File: WebDriverAgentLib/Routing/FBXCElementSnapshotWrapper.m (hunk 1 above)
Comment (KazuCocoa) [reply WebDriverAgentLib/Routing/FBXCElementSnapshotWrapper.m:107]: btw, the `selected` is also not clear. It was also always false in my some checks with focus. interesting..

**Pull request closed without merging**